Update conf for Molecule 3, apply ansible-lint

benformosa opened the pull request:
Fixes #144 
Updates molecule.yml with the updated lint syntax: https://molecule.readthedocs.io/en/latest/configuration.html#lint

Also various changes to tasks to pass ansible-lint rules

benformosa committed:
apply ansible-lint

benformosa committed:
update molecule lint configuration for v3

jamescassell reviewed (changes requested):
Some comments. I guess it'd require trawling git history to get the "why" for one of them.

benformosa committed:
remove unnecessary quotes

benformosa committed:
remove rules from ansible-lint skip list

I've resolved the issues causing rules 306 (require pipefail) and
602 (don't compare against empty string).

jamescassell reviewed (approved):
I guess I disagree with rule 602, but that's just my opinion.  Fix the last extra quote and I'll merge it, barring anyone protesting.

benformosa committed:
remove more unnecessary quotes
